LoopKit/GlucoseKit/GlucoseStore.swift

```diff
@@ -372,52 +370,57 @@ extension GlucoseStore {
     
     private func saveSamplesToHealthKit() {
         dispatchPrecondition(condition: .onQueue(queue))
-        var error: Error? = nil
+        var error: Error?
         
         cacheStore.managedObjectContext.performAndWait {
-            let request: NSFetchRequest<CachedGlucoseObject> = CachedGlucoseObject.fetchRequest()
-            request.predicate = NSPredicate(format: "healthKitEligibleDate <= %@", Date() as NSDate)
             do {
-                let stored = try self.cacheStore.managedObjectContext.fetch(request)
-                guard !stored.isEmpty else {
+                let request: NSFetchRequest<CachedGlucoseObject> = CachedGlucoseObject.fetchRequest()
+                request.predicate = NSPredicate(format: "healthKitEligibleDate <= %@", Date() as NSDate)
+                request.sortDescriptors = [NSSortDescriptor(key: "modificationCounter", ascending: true)]   // Maintains modificationCounter order
+
+                let objects = try self.cacheStore.managedObjectContext.fetch(request)
+                guard !objects.isEmpty else {
                     return
                 }
                 
-                if stored.contains(where: { $0.uuid != nil }) {
+                if objects.contains(where: { $0.uuid != nil }) {
                     self.log.error("Found CachedGlucoseObjects with non-nil uuid. Should never happen, but HealthKit should be able to resolve it.")
-                    // Note the UUIDs will be overwritten below...HealthKit should resolve this by "replacing" these items since the syncIdentifiers match
+                    // Note: UUIDs will be overwritten below, but since the syncIdentifiers will match then HealthKit can resolve correctly via replacement
                 }
                     
-                let quantitySamples = stored.map { $0.quantitySample }
+                let quantitySamples = objects.map { $0.quantitySample }
                 
-                // Save objects to HealthKit, log any errors, but do not fail
                 let dispatchGroup = DispatchGroup()
                 dispatchGroup.enter()
                 self.healthStore.save(quantitySamples) { (_, healthKitError) in
                     error = healthKitError
                     dispatchGroup.leave()
                 }
                 dispatchGroup.wait()
-                
-                let zipped = zip(quantitySamples, stored).map { ($0, $1) }
-                // Update Core Data with the changes, log any errors, but do not fail
-                zipped.forEach { quantitySample, object in
+
+                guard error == nil else {
```

Comment:
[nit] put comment here saying something like ("this ensures a HealthKit storage error will cause the uuid to not be updated")